### PR TITLE
Fix wazuh.yml link and in-file documentation

### PR DESCRIPTION
**Pull request by AlexRuiz7**
## Summary

This PR solves the issue noted in the issue https://github.com/wazuh/wazuh-kibana-app/issues/4349#issuecomment-1204050742.

## Result

```yml
---
#
# Wazuh app - App configuration file
# Copyright (C) 2015-2022 Wazuh, Inc.
#
# This program is free software; you can redistribute it and/or modify
# it under the terms of the GNU General Public License as published by
# the Free Software Foundation; either version 2 of the License, or
# (at your option) any later version.
#
# Find more information about this on the LICENSE file.
#
# ======================== Wazuh app configuration file ========================
#
# Please check the documentation for more information about configuration options:
# https://documentation.wazuh.com/4.3/user-manual/wazuh-dashboard/config-file.html
#
# Also, you can check our repository:
# https://github.com/wazuh/wazuh-kibana-app
#
# ---------------------------- Unauthorized roles ------------------------------
#
# Disable Wazuh for the Elasticsearch / OpenSearch roles defined here.
# disabled_roles:
#   - wazuh_disabled
#
# ------------------------------- Index patterns -------------------------------
#
# Default index pattern to use on the app. If there's no valid index pattern, the
# app will automatically create one with the name indicated in this option.
# pattern: wazuh-alerts-*
#
# ----------------------------------- Checks -----------------------------------
#
# Define which checks will be executed by the App's HealthCheck.
# Allowed values are: true, false
#
# Enable or disable the index pattern health check when opening the app.
# checks.pattern: true
#
# Enable or disable the template health check when opening the app.
# checks.template: true
#
# Enable or disable the API health check when opening the app.
# checks.api: true
#
# Enable or disable the setup health check when opening the app.
# checks.setup: true
#
# Enable or disable the known fields health check when opening the app.
# checks.fields: true
#
# Change the default value of the Kibana metaField configuration
# checks.metaFields: true
#
# Change the default value of the Kibana timeFilter configuration
# checks.timeFilter: true
#
# Change the default value of the Kibana max buckets configuration
# checks.maxBuckets: true
#
# --------------------------------- Extensions ---------------------------------
#
# Define the initial state of the extensions (enabled / disabled) for recently
# added hosts. The extensions can be enabled or disabled anytime using the UI.
# Allowed values are: true, false
#
# Enable or disable the PCI DSS tab on Overview and Agents.
# extensions.pci: true
#
# Enable or disable the GDPR tab on Overview and Agents.
# extensions.gdpr: true
#
# Enable or disable the HIPAA tab on Overview and Agents.
# extensions.hipaa: true
#
# Enable or disable the NIST 800-53 tab on Overview and Agents.
# extensions.nist: true
#
# Enable or disable the TSC tab on Overview and Agents.
# extensions.tsc: true
#
# Enable or disable the Audit tab on Overview and Agents.
# extensions.audit: true
#
# Enable or disable the Open SCAP tab on Overview and Agents.
# extensions.oscap: false
#
# Enable or disable the CIS-CAT tab on Overview and Agents.
# extensions.ciscat: false
#
# Enable or disable the Amazon (AWS) tab on Overview.
# extensions.aws: false
#
# Enable or disable the Google Cloud Platform tab on Overview.
# extensions.gcp: false
#
# Enable or disable the VirusTotal tab on Overview and Agents.
# extensions.virustotal: false
#
# Enable or disable the Osquery tab on Overview and Agents.
# extensions.osquery: false
#
# Enable or disable the Docker listener tab on Overview and Agents.
# extensions.docker: false
#
# ------------------------------- Timeout --------------------------------------
#
# Maximum time, in milliseconds, the app will wait for an API response when making
# requests to it. It will be ignored if the value is set under 1500 milliseconds.
# timeout: 20000
#
# --------------------------- Index pattern selector ---------------------------
#
# Define if the user is allowed to change the selected index pattern directly from
# the top menu bar.
# ip.selector: true
#
# Disable certain index pattern names from being available in index pattern
# selector from the Wazuh app.
# ip.ignore: 
#
# ------------------------------ Monitoring ------------------------------------
#
# Enable or disable the wazuh-monitoring index creation and/or visualization.
# wazuh.monitoring.enabled: true
#
# Frequency, in seconds, of API requests to get the state of the agents and create
# a new document in the wazuh-monitoring index with this data.
# wazuh.monitoring.frequency: 900
#
# Define the number of shards to use for the wazuh-monitoring-* indices.
# wazuh.monitoring.shards: 1
#
# Define the number of replicas to use for the wazuh-monitoring-* indices.
# wazuh.monitoring.replicas: 0
#
# Define the interval in which a new wazuh-monitoring index will be created.
# Allowed values are: h (hourly), d (daily), w (weekly), m (monthly)
# wazuh.monitoring.creation: w
#
# Default index pattern to use for Wazuh monitoring.
# wazuh.monitoring.pattern: wazuh-monitoring-*
#
# --------------------------------- Sample data --------------------------------
#
# Define the index name prefix of sample alerts. It must match the template used
# by the index pattern to avoid unknown fields in dashboards.
# alerts.sample.prefix: wazuh-alerts-4.x-
#
# ------------------------------ Background tasks ------------------------------
#
# Define the index prefix of predefined jobs.
# cron.prefix: wazuh
#
# ------------------------------ Wazuh Statistics ------------------------------
#
# Enable or disable the statistics tasks.
# cron.statistics.status: true
#
# Enter the ID of the hosts you want to save data from, leave this empty to run
# the task on every host.
# cron.statistics.apis: 
#
# Define the frequency of task execution using cron schedule expressions.
# cron.statistics.interval: 0 */5 * * * *
#
# Define the name of the index in which the documents will be saved.
# cron.statistics.index.name: statistics
#
# Define the interval in which a new index will be created.
# cron.statistics.index.creation: w
#
# Define the number of shards to use for the statistics indices.
# cron.statistics.shards: 1
#
# Define the number of replicas to use for the statistics indices.
# cron.statistics.replicas: 0
#
# ------------------------------ Logo customization ----------------------------
#
# Set the name of the app logo stored at /plugins/wazuh/public/assets/
# customization.logo.app: 
#
# Set the name of the sidebar logo stored at /plugins/wazuh/public/assets/
# customization.logo.sidebar: 
#
# Set the name of the health-check logo stored at /plugins/wazuh/public/assets/
# customization.logo.healthcheck: 
#
# Set the name of the reports logo (.png) stored at /plugins/wazuh/public/assets/
# customization.logo.reports: 
#
# ---------------------------- Hide manager alerts -----------------------------
#
# Hide the alerts of the manager in every dashboard.
# hideManagerAlerts: false
#
# ------------------------------- App logging level ----------------------------
#
# Logging level of the App.
# Allowed values are: info, debug
# logs.level: info
#
# ------------------------------- Agent enrollment -----------------------------
#
# Specifies the Wazuh registration server, used for the agent enrollment.
# enrollment.dns: 
#
# Specifies the password used to authenticate during the agent enrollment.
# enrollment.password: 
#
#-------------------------------- Wazuh hosts ----------------------------------
#
# The following configuration is the default structure to define a host.
#
# hosts:
# # Host ID / name,
#   - env-1:
#     # Host URL
#     url: https://env-1.example
#     # Host / API port
#     port: 55000
#     # Host / API username
#     username: wazuh-wui
#     # Host / API password
#     password: wazuh-wui
#     # Use RBAC or not. If set to true, the username must be "wazuh-wui".
#     run_as: true
#   - env-2:
#     url: https://env-2.example
#     port: 55000
#     username: wazuh-wui
#     password: wazuh-wui
#     run_as: true

hosts:
  - default:
    url: https://localhost
    port: 55000
    username: wazuh-wui
    password: wazuh-wui
    run_as: false
```

## Test

- The `wazuh.yml` file is created without errors.
- The link to the web documentation is right: https://documentation.wazuh.com/4.3/user-manual/wazuh-dashboard/config-file.html
- The in-file documentation is understandable and free of errors (of any kind).
- The in-file documentation is consistent with the UI and web documentation.

### CODE REVIEW

File: server/lib/initial-wazuh-config.ts [hunk 1]
```diff
@@ -27,211 +63,226 @@ export const initialWazuhConfig: string = `---
...
-     port: 55000
-     username: wazuh-wui
-     password: wazuh-wui
-     run_as: false
-`
+    url: https://localhost
+    port: 55000
+    username: wazuh-wui
+    password: wazuh-wui
+    run_as: false
```
Comment (Desvelao) [inline server/lib/initial-wazuh-config.ts:287]: **issue**: I think there is a problem with the identation of these settings that causes an error and when accessing the Wazuh plugin I get:
image

**suggestion**: Add the spaces that were removed.

If we check this is the problem, then it should modify the examples above too.

Comment (AlexRuiz7) [reply server/lib/initial-wazuh-config.ts]: The indentation problem has been fixed in https://github.com/wazuh/wazuh-kibana-app/pull/4378/commits/00b9f037dadbb727065f7da07fd6777941244b23.

Please re-review the changes.

Thank you!

**Review by Desvelao — CHANGES_REQUESTED** [review]
**issue**: the object that contains the default configuration could be muted, so changes in the configuration could do the plugin behaves as not expected ways.

The `defaultConfig` variable should be a new object, so the changes in its properties doesn't affect to the object that contains the default configuration `WAZUH_DEFATUL_APP_CONFIG`.

For this case, taking into account that the properties are in the first level, we could use the spread operator to "copy" the properties.
```ts
const defaultConfig = { ...WAZUH_DEFATUL_APP_CONFIG };
```

**Comment by AlexRuiz7** [comment]
Solved in https://github.com/wazuh/wazuh-kibana-app/pull/4378/commits/4bb051930941e12cf430a1366692d8f948ea9658

**Review by Desvelao — APPROVED** [review]
**review**
Code review :heavy_check_mark: 
Test :heavy_check_mark:

**Review by Machi3mfl — APPROVED** [review]
CR ✅
Test ✅

**Pull request merged**